Python/pylifecycle.c

```diff
@@ -1287,11 +1292,9 @@ init_interp_main(PyThreadState *tstate)
         }
     }
 
-    if ((interp->rare_events.builtins_dict_watcher_id = PyDict_AddWatcher(&builtins_dict_watcher)) == -1) {
-        return _PyStatus_ERR("failed to add builtin dict watcher");
-    }
 
-    if (PyDict_Watch(interp->rare_events.builtins_dict_watcher_id, interp->builtins) != 0) {
+    interp->dict_state.watchers[0] = &builtins_dict_watcher;
```

Comment:
Since the first two slots are "reserved", it might be clearer to have `#define`s for these slots. e.g. `BUILTINS_DICT_WATCHER_ID` for 0 here.

Reply:
What is the rationale for bypassing `PyDict_AddWatcher` here and special-casing?  Just a small amount of startup performance?  Here, consistency and encapsulating management of the `dict_state.watchers` array might be better maintenance-wise.

Reply:
It keeps things simpler if we are guaranteed to have IDs. Otherwise we need to worry about what happens if we can't add either or both of the watchers.